tests/data/test_artifact_dataset_sources.py

```diff
@@ -0,0 +1,121 @@
+import json
+import os
+
+import pytest
+from unittest import mock
+
+from mlflow.data.filesystem_dataset_source import FileSystemDatasetSource
+from mlflow.data.dataset_source_registry import resolve_dataset_source, get_dataset_source_from_json
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+
+
+@pytest.mark.parametrize(
+    ("source_uri", "source_type", "source_class_name"),
+    [
+        ("/tmp/path/to/my/local/file.txt", "local", "LocalArtifactDatasetSource"),
+        ("file:///tmp/path/to/my/local/directory", "local", "LocalArtifactDatasetSource"),
+        ("s3://mybucket/path/to/my/file.txt", "s3", "S3ArtifactDatasetSource"),
+        ("gs://mybucket/path/to/my/dir", "gs", "GCSArtifactDatasetSource"),
+        ("wasbs://user@host.blob.core.windows.net/dir", "wasbs", "AzureBlobArtifactDatasetSource"),
+        ("ftp://mysite.com/path/to/my/file.txt", "ftp", "FTPArtifactDatasetSource"),
+        ("sftp://mysite.com/path/to/my/dir", "sftp", "SFTPArtifactDatasetSource"),
+        ("hdfs://host_name:8020/hdfs/path/to/my/file.txt", "hdfs", "HdfsArtifactDatasetSource"),
+        ("viewfs://host_name:8020/path/to/my/dir", "viewfs", "HdfsArtifactDatasetSource"),
```

Comment:
Should this say `ViewFsArtifactDatasetSource`?

Reply:
`HdfsArtifactRepository` implements artifact operations for hdfs:/ and viewfs:/ locations, so `HdfsArtifactDatasetSource` is appropriate here.